framework/test/unit/modules/interoperability/base_cross_chain_update_command.spec.ts

```diff
@@ -792,85 +796,149 @@ describe('BaseCrossChainUpdateCommand', () => {
+				const ccm = {
+					crossChainCommand: CROSS_CHAIN_COMMAND_REGISTRATION,
+					fee: BigInt(0),
+					module: MODULE_NAME_INTEROPERABILITY,
+					nonce: BigInt(1),
+					params: utils.getRandomBytes(10),
+					// must be same as `context.chainID` to pass `!context.chainID.equals(ccm.receivingChainID)` check
+					receivingChainID: chainID,
+					// this will fail for `!ccm.sendingChainID.equals(params.sendingChainID)`
+					// params.sendingChainID is `defaultSendingChainID` (line 158)
+					sendingChainID: Buffer.from([1, 2, 3, 4]),
+					status: CCMStatusCode.OK,
+				};
+
+				executeContext = createTransactionContext({
+					chainID,
+					stateStore,
+					transaction: new Transaction({
+						...defaultTransaction,
+						command: command.name,
+						params: codec.encode(crossChainUpdateTransactionParams, {
+							...params,
+							inboxUpdate: {
+								...params.inboxUpdate,
+								crossChainMessages: [codec.encode(ccmSchema, ccm)],
+							},
+						}),
+					}),
+				}).createCommandExecuteContext(command.schema);
+
+				try {
+					command['verifyRoutingRules'](ccm, executeContext.params, executeContext.chainID, true);
+				} catch (err: any) {
+					expect((err as Error).message).toBe('CCM is not from the sending chain.');
+				}
+			});
+
+			it('should return error when ccm status is CCMStatusCode.CHANNEL_UNAVAILABLE', () => {
+				const ccm = {
+					crossChainCommand: CROSS_CHAIN_COMMAND_REGISTRATION,
+					fee: BigInt(0),
+					module: MODULE_NAME_INTEROPERABILITY,
+					nonce: BigInt(1),
+					params: utils.getRandomBytes(10),
+					// must be same as `context.chainID` to pass `!context.chainID.equals(ccm.receivingChainID)`
+					receivingChainID: chainID,
+					// must be same as defaultSendingChainID to pass `!ccm.sendingChainID.equals(params.sendingChainID)`
+					sendingChainID: defaultSendingChainID,
+					// will fail for `CCMStatusCode.CHANNEL_UNAVAILABLE`
+					status: CCMStatusCode.CHANNEL_UNAVAILABLE, // ***
+				};
+
+				executeContext = createTransactionContext({
+					chainID,
+					stateStore,
+					transaction: new Transaction({
+						...defaultTransaction,
+						command: command.name,
+						params: codec.encode(crossChainUpdateTransactionParams, {
+							...params,
+							inboxUpdate: {
+								...params.inboxUpdate,
+								crossChainMessages: [codec.encode(ccmSchema, ccm)],
+							},
+						}),
+					}),
+				}).createCommandExecuteContext(command.schema);
+
+				try {
+					command['verifyRoutingRules'](ccm, executeContext.params, executeContext.chainID, true);
+				} catch (err: any) {
+					expect((err as Error).message).toBe(
+						'CCM status channel unavailable can only be set on the mainchain.',
+					);
+				}
+			});
 		});
 
-		it('should terminate the chain and add an event when CCM is not directed to the sidechain', async () => {
+		it('should return error when CCM is not directed to the sidechain', () => {
```

Comment:
This can go under another `describe('sidechain')` following the above test suite.

Reply:
@sitetester this comment is still not resolved

Reply:
> It is under describe('verifyRoutingRules' (but not describe('isMainchain')

True, you were right. But also good in the current way. Can be resolved for me.